### PR TITLE
Fix vacancies without minimum value

### LINKED ISSUE #206: Vacancies minimum value
Add a validator to the `vacancies` field to be a minimum of 1

### PR DESCRIPTION
**Pull request by BrunoRosendo**
closes #206

### CODE REVIEW

File: test/utils/ValidatorTester.js [hunk 1]
```diff
@@ -332,6 +332,26 @@ const ValidatorTester = (requestEndpoint) => (location) => (field_name) => ({
             });
         });
     },
+
+    mustBeGreaterThanOrEqualTo: (min) => {
```
Comment (DoStini) [inline test/utils/ValidatorTester.js:336]: Don't we have any case where we test the max one?

Comment (BrunoRosendo) [reply test/utils/ValidatorTester.js:336]: I don't think so. We don't have any case for that one, so I don't think we should add the code. If necessary, It's very easy to create it by looking at this one

File: src/models/Offer.js [hunk 2]
```diff
@@ -62,7 +62,7 @@ const OfferSchema = new Schema({
     },
 
     isPaid: { type: Boolean },
-    vacancies: { type: Number },
+    vacancies: { type: Number, min: OfferConstants.vacancies.min },
```
Comment (imnotteixeira) [inline src/models/Offer.js:65]: Does this allow an empty value as well?

File: package.json [hunk 3]
```diff
@@ -48,7 +48,7 @@
     "morgan": "^1.9.1",
     "multer": "^1.4.2",
     "nodemailer": "^6.7.2",
-    "nodemailer-express-handlebars": "^4.0.0",
+    "nodemailer-express-handlebars": "^5.0.0",
```
Comment (imnotteixeira) [inline package.json:51]: we did it, nice :P

File: src/models/Offer.js (hunk 2 above)
Comment (BrunoRosendo) [reply src/models/Offer.js:65]: I took this form the docs:
`Validators are not run on undefined values. The only exception is the [required validator](https://mongoosejs.com/docs/api.html#schematype_SchemaType-required).` 

[here](https://mongoosejs.com/docs/schematypes.html#number-validators)

File: src/models/constants/Offer.js [hunk 4]
```diff
@@ -12,6 +12,9 @@ export default Object.freeze({
     requirements: {
         min_length: 1,
     },
+    vacancies: {
+        min: 1,
```
Comment (imnotteixeira) [inline src/models/constants/Offer.js:16]: It should probably be allowed to be 0. Thinking for the future if we ever update offers' vacancies, they can end up with 0 vacancies. With the current code, the DB wouldn't let that.

**Review by imnotteixeira — CHANGES_REQUESTED** [review]

File: src/models/constants/Offer.js (hunk 4 above)
Comment (BrunoRosendo) [reply src/models/constants/Offer.js:16]: Should I update only the DB or the validators as well?

Comment (imnotteixeira) [reply src/models/constants/Offer.js:16]: DB at least, validators if you want

**Review by imnotteixeira — APPROVED** [review]

**Pull request merged**